src/datahandlers/kegg.py

```diff
@@ -18,111 +20,152 @@
 # Which can be parsed to find the KEGG compounds that have a sequence.
 # As for crawling them and pulling the sequence, should we be going through the KEGG client? probably?
 
+
 def pull_kegg_compound_labels(outfile):
-    request_session = requests.Session()
-    request_adapter = requests.adapters.HTTPAdapter(max_retries=10)
-    request_session.mount('http://', request_adapter)
-    request_session.mount('https://', request_adapter)
-    with open(outfile,'w') as lfile:
-        for i in range(1,22250):
-            rid = f'C{str(i).zfill(5)}'
-            url = f'http://rest.kegg.jp/get/cpd:{rid}'
-            raw_results = request_session.get(url)
-            rawlines = raw_results.text.split('\n')
-            if len(rawlines) > 0:
-                if rawlines[0].startswith('ENTRY'):
-                    if rawlines[1].startswith('NAME'):
-                        name = ' '.join(rawlines[1].split()[1:])
-                        if name.endswith(';'):
+    try:
+
+        request_session = requests.Session()
+        request_adapter = requests.adapters.HTTPAdapter(max_retries=10)
+        request_session.mount("http://", request_adapter)
+        request_session.mount("https://", request_adapter)
+
+        # 1:22530 + 1 to have inclusive end
+        compound_chunks = list(chunked(range(1, 22531), 10))
```

Comment:
I think this is a good opportunity to document why we specifically get CPDs from 1:22530 here 😄 . Do you know the answer to that? If not, we can ask @cbizon.

Reply:
I don't know the answer to this, but I can speculate.  I believe it's that the entries are added & the collection grows.  Previous commit was a range from 1:22250 & when I was looking the kegg api, no more data was available after 22530...indicating that this a moving target.  I could not find an endpoint that specified the current range such that we don't have to hard code it.

Reply:
Unfortunately, there's no way to know the right answer here.  There's not an API or anything that lets you know this number.  So that's an experimentally determined value, and potentially a moving target.   KEGG used to make data tables available for download, but stopped that practice a number of years ago.